canister/src/unstable_blocks.rs

```diff
@@ -167,6 +167,24 @@ impl UnstableBlocks {
         chain.reverse();
         chain
     }
+
+    /// Returns block headers in the inclusive range provided as an argument
+    /// relative to the start of unstable blocks.
+    pub(crate) fn get_block_headers_in_range(
+        &self,
+        start_range_in_unstable_blocks: u32,
+        end_range_in_unstable_blocks: u32,
+    ) -> Vec<Vec<u8>> {
```

Comment:
Two comments here:

1. Like all other methods in this component, we should return `BlockHeader` and not necessarily assume that the caller wants it to be encoded. This function maybe used for other purposes in the future.
2. I'd find it more intuitive if we let this method take the actual heights and not the unstable heights. The unstable heights can be computed within this function by passing in the `stable_height`. We already do this for other methods in this component.

The method signature I suggested addresses these two issues.

```
   pub fn get_block_headers_in_range(
        &self,
        stable_height: Height,
        heights: std::ops::RangeInclusive<Height>,
    ) -> impl Iterator<Item = &BlockHeader> {
```

Reply:
Done.